clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3006,6 +3006,49 @@ static void handleNumSimdWorkItemsAttr(Sema &S, Decl *D,
                                                                      E);
 }
 
+// Add scheduler_target_fmax_mhz
+void Sema::addSYCLIntelSchedulerTargetFmaxMhzAttr(
+    Decl *D, const AttributeCommonInfo &Attr, Expr *E) {
+  if (!E)
+    return;
+
+  if (!E->isInstantiationDependent()) {
+    Optional<llvm::APSInt> ArgVal = E->getIntegerConstantExpr(getASTContext());
+    if (!ArgVal) {
+      Diag(E->getExprLoc(), diag::err_attribute_argument_type)
+          << Attr.getAttrName() << AANT_ArgumentIntegerConstant
+          << E->getSourceRange();
+      return;
+    }
+    uint32_t TargetFmaxMhz = ArgVal->getSExtValue();
+    uint32_t UpperLimit = 1048576;
+    if (TargetFmaxMhz > UpperLimit) {
```

Comment:
Can we re-use this function here https://github.com/intel/llvm/blob/a7ad8b8caab43ffe3540dc94692859c0ad3130e8/clang/lib/Sema/SemaDeclAttr.cpp#L3880
NOTE: you probably need to define `getMinValue` and `getMaxValue` functions for the new attribute. For example like here https://github.com/intel/llvm/blob/a7ad8b8caab43ffe3540dc94692859c0ad3130e8/clang/include/clang/Basic/Attr.td#L1909

Reply:
Yes. Done. Thanks!